spec/Choices/CatalogSpec.php

```diff
@@ -0,0 +1,171 @@
+
+    function let(): void
+    {
+        $this->beConstructedWith($this->testDir, $this->templatesDir);
+    }
+
+    function it_is_initializable(): void
+    {
+        $this->shouldHaveType(Catalog::class);
+    }
+
+    function it_implements_catalog_interface(): void
+    {
+        $this->shouldHaveType(CatalogInterface::class);
+    }
+
+    function it_return_default_template_if_directory_doesnt_exist(
+        Finder $finder
+    ): void
+    {
+        $finder->files()->willReturn($finder);
+        $finder->in(Argument::type('string'))->willReturn($finder);
+        $finder->name('*.html.twig')->willReturn($finder);
+        $finder->depth(0)->willThrow(DirectoryNotFoundException::class);
+
+        $this->getTemplates()->shouldReturn(self::DEFAULT_TEMPLATE);
+    }
+
+    function it_return_default_template_if_directory_doesnt_contain_twig_files(
+        Finder $finder
+    ): void
+    {
+        $dir = $this->testDir.'/templates/'.$this->templatesDir;
+        if (!is_dir($dir)) {
+            mkdir($dir, 0777, true);
+        }
+        $finder->files()->willReturn($finder);
+        $finder->in($dir)->willReturn($finder);
+        $finder->name('*.html.twig')->willReturn($finder);
+        $finder->depth(0)->willReturn($finder);
+
+        $finder->hasResults()->willReturn(false);
+
+        $this->getTemplates()->shouldReturn(self::DEFAULT_TEMPLATE);
+
+        $this->rrmdir($this->testDir);
+    }
+
+    function it_return_templates_from_default_dir(
+        Finder $finder,
+        SplFileInfo $file1,
+        SplFileInfo $file2
+    ): void
+    {
+        $dir = $this->testDir.'/templates/'.$this->templatesDir;
+        if (!is_dir($dir)) {
+            mkdir($dir, 0777, true);
+        }
+        $finder->files()->willReturn($finder);
+        $finder->in($dir)->willReturn($finder);
+        $finder->name('*.html.twig')->willReturn($finder);
+        $finder->depth(0)->willReturn($finder);
+        $finder->hasResults()->willReturn(true);
+
+
+        $finder->getIterator()->willReturn(new \ArrayIterator([
+            $file1->getWrappedObject(),
+            $file2->getWrappedObject(),
+        ]));
+
+        $file1->getBasename('.html.twig')->willReturn('one');
+        $file1->getBasename()->willReturn('one.html.twig');
+        file_put_contents($dir.'/one.html.twig', '');
+
+        $file2->getBasename('.html.twig')->willReturn('two');
+        $file2->getBasename()->willReturn('two.html.twig');
+        file_put_contents($dir.'/two.html.twig', '');
+
+        $templates = [
+            'two' => $this->templatesDir.'/two.html.twig',
+            'one' => $this->templatesDir.'/one.html.twig',
+        ];
+
+        $this->getTemplates()->shouldReturn(array_merge(self::DEFAULT_TEMPLATE, $templates));
+
+        $this->rrmdir($this->testDir);
+    }
+    function it_return_templates_from_default_dir_with_first_default_if_default_exist(
```

Comment:
In my opinion spec here is too long. It seems the logic could be extracted into separate class.

Reply:
Additionally, it "returns", not "return" :)